src/entrypoint.sh

```diff
@@ -155,20 +169,54 @@ function git_init() {
     mkdir -p "${HOME}"/.ssh
     ssh-keyscan -t rsa "${source_repo_hostname}" >> "${HOME}"/.ssh/known_hosts
   else
-    info "the source repository is located within GitHub."               
-    if [[ -n "${SOURCE_GH_TOKEN}" ]]; then      
-      unset GITHUB_TOKEN      
-      gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${SOURCE_GH_TOKEN}"        
-      if [[ -n "${TARGET_GH_TOKEN}" ]]; then
-        gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${TARGET_GH_TOKEN}"
-      fi            
-      gh auth switch      
-      gh auth setup-git --hostname "${source_repo_hostname}"
-      info "done set git global configuration"    
-    else
-      info "default token to be used" 
-      gh auth setup-git --hostname "${source_repo_hostname}"      
-    fi       
+    info "the source repository is located within GitHub."
+  fi
+  echo "::endgroup::"
+}
+
+#######################################
+# doing the login to the source repository using gh cli
+# Arguments:
+#   source_repo_hostname
+#######################################
+function gh_login_src_github() {
+  echo "::group::login src github"
+  local source_repo_hostname=$1
+  # GITHUB_TOKEN is deprecated and can be removed in the future
+  if [[ -n "${SOURCE_GH_TOKEN}" ]] || [[ -n "${GITHUB_TOKEN}" ]]; then
+    ################################
+    if [[ -n "${GITHUB_TOKEN}" ]]; then
+      warn "github_token parameter is deprecated please use source_gh_token."
+      info "setting SOURCE_GH_TOKEN"
+      export SOURCE_GH_TOKEN="${GITHUB_TOKEN}"
+      unset GITHUB_TOKEN
+    fi
+    ###############################
+    if [[ -z "${SOURCE_GH_TOKEN}" ]]; then
+      err "Missing input 'source_gh_token: \${{ secrets.GITHUB_TOKEN }}'.";
+      exit 1;
+    fi
+    info "source server url: ${source_repo_hostname}"
+    info "logging out"
+    gh auth logout --hostname "${source_repo_hostname}" || debug "not logged in"
+    info "login to the source git repository"
+    gh auth login --git-protocol "https" --hostname "${source_repo_hostname}" --with-token <<< "${SOURCE_GH_TOKEN}"
+    gh auth status
+    # if [[ -n "${TARGET_GH_TOKEN}" ]]; then
```

Comment:
the `gh auth switch` can be a bit dangerous because we do not specify the user. 
Therefore I think to logout is a better approach.

Moved the target login into a seperate part